zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/PermissionsBehavior.java

```diff
@@ -108,7 +105,7 @@ public Either<Rejection, AuthorizationRecord> permissionDoesNotExist(
       final AuthorizationRecord record) {
     for (final PermissionValue permission : record.getPermissions()) {
       final var currentResourceIdentifiers =
-          authCheckBehavior.getAuthorizedResourceIdentifiers(
+          authCheckBehavior.getAllAuthorizedResourceIdentifiers(
```

Comment:
💭 This should actually become direct as well. When we remove permissions from an entity we should make sure it's actually assigned to this entity directly. Do you want to take it as part of this, or should I create a separate issue for it?

Reply:
Ah okay, I thought about it and came to the opposite conclusion: that we would allow deleting an inherited permission. But makes sense that we don't support that and only allow deleting on the entity directly.

Reply:
It's about expectations I suppose. As a user if I want to remove permission X from user Y I wouldn't expect that user Z would also lose the same permission.

Reply:
Oh right, that could easily cause mistakes, right :+1:

Reply:
Fixed in b8db2fd